 # <font color=darkblue>Programming for Data Analysis 2022 - Project 2</font>

## <font color=darkblue>Orla Corry</font>
***

## <font color=darkblue>Table of Contents</font>
***

- [Introduction](#Introduction)


- [Importing Libraries](#Importing-Libraries)


- [Reading in the csv file](#Reading-in-the-csv-file)


- [Analysis/review of the dataset](#Analysis/review-of-the-dataset)


- [References](#References)

## <font color=darkblue>Introduction</font>
***

## <font color=darkblue>Importing Libraries</font>
***

For this project I will be importing the following Libraries:

1. **Pandas** - to read in the dataset and to analyse data that is in tabular form

In [2]:
import pandas as pd

## <font color=darkblue>Reading in the csv file</font>
***
I have sourced the Wesconsin Breast Cancer Dataset from <a href="https://github.com/jeffheaton/aifh/blob/master/vol1/python-examples/datasets/breast-cancer-wisconsin.csv" target="_top">Heaton. J (Github), 2013</a>. To pull the actual csv URL from Heaton's Github, I clicked on the *Raw* tab and copied & pasted the URL that was provided for the raw data. Below, I have read in the the csv file to Python using the built in library <a href="https://pandas.pydata.org/docs/" target="_top">Pandas</a>.


In [13]:
# Reading in the csv file using pandas
# Using dataset from https://github.com/jeffheaton/aifh/blob/master/vol1/python-examples/datasets/breast-cancer-wisconsin.csv
dataset = ("https://raw.githubusercontent.com/jeffheaton/aifh/master/vol1/python-examples/datasets/breast-cancer-wisconsin.csv")
df = pd.read_csv (dataset) #storing the read in csv dataframe as df 

In [16]:
# Printing the dataframe (df) to make sure the csv file read in to Python properly
df
#df.shape

id  clump_thickness  size_uniformity  shape_uniformity  \
0    1000025                5                1                 1   
1    1002945                5                4                 4   
2    1015425                3                1                 1   
3    1016277                6                8                 8   
4    1017023                4                1                 1   
..       ...              ...              ...               ...   
694   776715                3                1                 1   
695   841769                2                1                 1   
696   888820                5               10                10   
697   897471                4                8                 6   
698   897471                4                8                 8   

     marginal_adhesion  epithelial_size bare_nucleoli  bland_chromatin  \
0                    1                2             1                3   
1                    5                7            10                3   
2                    1                2             2                3   
3                    1                3             4                3   
4                    3                2             1                3   
..                 ...              ...           ...              ...   
694                  1                3             2                1   
695                  1                2             1                1   
696                  3                7             3                8   
697                  4                3             4               10   
698                  5                4             5               10   

     normal_nucleoli  mitoses  class  
0                  1        1      2  
1                  2        1      2  
2                  1        1      2  
3                  7        1      2  
4                  1        1      2  
..               ...      ...    ...  
694                1        1      2  
695                1        1      2  
696               10        2      4  
697                6        1      4  
698                4        1      4  

[699 rows x 11 columns]

## <font color=darkblue>Analysis/review of the dataset</font>
***

Using the <a href="https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.shape.html" target="_top">df.shape</a> command on the dataframe above, we can see that the dataset comprises of 699 instances/cases (rows) of data across 11 attributes (columns).

## <font color=darkblue>References</font>
***

1. Heaton J. Github. breast-cancer-wisconsin.csv (2013) https://github.com/jeffheaton/aifh/blob/master/vol1/python-examples/datasets/breast-cancer-wisconsin.csv


2. Pandas Documentation. (2022). https://pandas.pydata.org/docs/


3. Pandas. pandas.DataFrame.shape. (2022). https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.shape.html